### DISCUSSION TITLE
Nested pydantic class in response_model

### DISCUSSION THREAD

**Discussion by lana-questions** [fastapi/fastapi#6951] [Questions] [answered]
### Example

Here's a self-contained, [minimal, reproducible, example](https://stackoverflow.com/help/minimal-reproducible-example) with my use case:

```Python
from fastapi import FastAPI
from pydantic.main import BaseModel


class Level2CustomInit(BaseModel):
    a: int
    b: int

    def __init__(self, number):
        super().__init__(a=number, b=number)


class Level2(BaseModel):
    number: int


class Level1(BaseModel):
    level2: Level2

    def __init__(self, number):
        _level2 = Level2(number=number)
        super().__init__(level2=_level2)


class Level1CustomInit(BaseModel):
    level2: Level2CustomInit

    def __init__(self, number):
        _level2 = Level2CustomInit(number=number)
        super().__init__(level2=_level2)


app = FastAPI()


@app.get("/level1", response_model=Level1)
async def func1(number: int):
    return {"number": number}


@app.get("/level1custominit", response_model=Level1CustomInit)
async def func1(number: int):
    return {"number": number}


if __name__ == "__main__":
    l1 = Level1(number=8)
    print("level 1", l1)

    l1 = Level1CustomInit(number=8)
    print("level 1 with custom init", l1)

    import uvicorn

    uvicorn.run("test:app", host="127.0.0.1", port=5000, log_level="info")

    # Calling API with http://localhost:5000/level1?number=3

    # {"a":3,"b":3}


    # Calling API with http://localhost:5000/level1custominit?number=3

    # pydantic.error_wrappers.ValidationError: 1 validation error for Level1
    # response -> level2
    #   __init__() got an unexpected keyword argument 'a' (type=type_error)
```

### Description

I want to use a pydantic model using a nested class with a custom __init__ function (Level1CustomInit with Level2CustomInit nested). I can construct a pydantic object with the custom constructor (so I assume it is not a problem of pydantic), but using this class as a response_model runs into a pydantic.error_wrappers.ValidationError. This error does not occur for a nested model without a custom initialization function (Level1 with Level2 nested).

Thanks a lot for your help!

### Environment

* OS: macOS
* FastAPI Version 0.62.0
* Python version: 3.8

**Comment by lana-questions** [comment, marked as answer]
The problem is solved when I don't define the Level2 object, but pass a dictionary with its initialization arguments.

```Python
class Level1CustomInit(BaseModel):
    level2: Level2CustomInit

    def __init__(self, number):
        super().__init__(level2={"number":number})
```

**Comment by ycd** [comment]
If your problem is resolved, do you mind closing the issue?

**Comment by tiangolo** [comment]
Thanks for the help here @ycd ! 👏  🙇 

Thanks for reporting back and closing the issue @lana-questions  👍

> Sorry for the long delay! 🙈  I wanted to personally address each issue/PR and they piled up through time, but now I'm checking each one in order.
